### PR TITLE
py-ptpython: new port (version 3.0.23); ptpython: obsolete

### PR DESCRIPTION
**Pull request by catap**
#### Description

Here an update of `ptpython` and moving it to usual python port format with per python version subport.

###### Type(s)

- [x] enhancement

###### Tested on

macOS 12.6.8 21G725 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-ptpython/Portfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+checksums               rmd160  88d712e3864658351a0d89b559d8de6aa1681908 \
+                        sha256  9fc9bec2cc51bc4000c1224d8c56241ce8a406b3d49ec8dc266f78cd3cd04ba4 \
+                        size    68146
+
+python.versions         38 39 310 311
+
+python.pep517           yes
+
+python.link_binaries    no
```
Comment (reneeotten) [inline python/py-ptpython/Portfile:29]: this is uncommon; why would you want this switched to no? I would expect this to be "yes" and then there would ideally no need for the "ln" stuff in the  "post-destroot"?

Comment (catap) [reply python/py-ptpython/Portfile:29]: @reneeotten to compatibility with layout that is created by current `ptpython` port.

Comment (reneeotten) [reply python/py-ptpython/Portfile:29]: IMO this should just behave as any other py-XY port. The binaries should go into ${prefix}/bin with the version suffixed. To select whatever version the user wants, one can use the "select" port. 

But perhaps I am not understanding it correctly...?

Comment (catap) [reply python/py-ptpython/Portfile:29]: @reneeotten right now it creates a layout:
```
/opt/local/bin/ptipython
/opt/local/bin/ptipython3
/opt/local/bin/ptipython3.10
/opt/local/bin/ptpython
/opt/local/bin/ptpython3
/opt/local/bin/ptpython3.10
```

So, I've moved `ptpython` and `ptpython3` to select's group and construct `ptpython3.10` by hand.

Comment (catap) [reply python/py-ptpython/Portfile:29]: @reneeotten anyway, if you think that it is not nessasary I gladly remove that hack.

Comment (catap) [reply python/py-ptpython/Portfile:29]: To make things simpler => I've removed that ugly hack. If anyone complains, will do something, but not yet.

File: python/ptpython/Portfile [hunk 2]
```diff
@@ -1,32 +1,17 @@
...
-                    port:py${python.version}-setuptools
+replaced_by         py-ptpython
+categories          python
+license             BSD
 
-depends_lib-append  port:py${python.version}-appdirs \
-                    port:py${python.version}-jedi \
-                    port:py${python.version}-prompt_toolkit \
-                    port:py${python.version}-pygments
+# remove after September 10, 202
```
Comment (mascguy) [inline python/ptpython/Portfile:17]: Minor nit-pick: Add the full year (2024).  ;-)

Comment (catap) [reply python/ptpython/Portfile]: @mascguy here it is!

**Review by mascguy — APPROVED** [review]

**Pull request merged**